src/file_organizer/parallel/processor.py

```
@@ -320,26 +336,69 @@ def finalize_result(file_result: FileResult) -> FileResult:
                 pending.remove(future)
                 path = future_paths.pop(future)
                 future_started.pop(future, None)
+                future_queued_at.pop(future, None)
+                # A healthy completion means the pool just made progress: a
+                # worker slot is now free for queued work. Reset the
+                # saturation-detection clock for every still-pending future so
+                # the 2×timeout guard only fires after *continuous* no-progress
```

Ruff (0.15.17)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)